Actually we seem to have a problem locally here under some circumstances:

```
 SecurityUtil.runAs(securityService, organization, user, () -> {
        try {
          conflictingEvents.addAll(persistence.getEvents(captureDeviceID, startDate, endDate, Util.EVENT_MINIMUM_SEPARATION_MILLISECONDS)
            .parallelStream().map(this::getEventMediaPackage).collect(Collectors.toList()));
```

where we're getting an exception from one of the threads created by the parallelStream():

```
Caused by: java.lang.IllegalStateException: No organization is set in security context
	at org.opencastproject.kernel.security.SecurityServiceSpringImpl.getUser(SecurityServiceSpringImpl.java:94) ~[?:?]
	at org.opencastproject.assetmanager.impl.AssetManagerWithSecurity.isAdmin(AssetManagerWithSecurity.java:229) ~[?:?]
	at org.opencastproject.assetmanager.impl.AssetManagerWithSecurity.access$000(AssetManagerWithSecurity.java:61) ~[?:?]
	at org.opencastproject.assetmanager.impl.AssetManagerWithSecurity$1.select(AssetManagerWithSecurity.java:138) ~[?:?]
	at org.opencastproject.scheduler.impl.SchedulerServiceImpl.getEventMediaPackage(SchedulerServiceImpl.java:1526) ~[?:?]
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193) ~[?:?]
	at java.util.Vector$VectorSpliterator.forEachRemaining(Vector.java:1440) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472) ~[?:?]
	at java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:747) ~[?:?]
	at java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:721) ~[?:?]
	at java.util.stream.AbstractTask.compute(AbstractTask.java:327) ~[?:?]
	at java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:731) ~[?:?]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289) ~[?:?]
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056) ~[?:?]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692) ~[?:?]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157) ~[?:?]
```

so looks like the approach used here in unsafe in some way, as the SecurityUtil.runAs() method is setting the user/organization in threadlocal for the current thread, but getEventMediaPackage is called in another thread which doesn't have a user/organization set.

--- a/modules/scheduler-impl/src/main/java/org/opencastproject/scheduler/impl/SchedulerServiceImpl.java
+++ b/modules/scheduler-impl/src/main/java/org/opencastproject/scheduler/impl/SchedulerServiceImpl.java
@@ -1088,7 +1088,7 @@ public class SchedulerServiceImpl extends AbstractIndexProducer implements Sched
       SecurityUtil.runAs(securityService, organization, user, () -> {
         try {
           conflictingEvents.addAll(persistence.getEvents(captureDeviceID, startDate, endDate, Util.EVENT_MINIMUM_SEPARATION_MILLISECONDS)
-            .parallelStream().map(this::getEventMediaPackage).collect(Collectors.toList()));
+            .stream().map(this::getEventMediaPackage).collect(Collectors.toList()));
         } catch (SchedulerServiceDatabaseException e) {
           logger.error("Failed to get conflicting events", e);
         }

